pkg/validation/validate_component.go

```diff
@@ -47,75 +50,108 @@ func NewComponentValidator(c []*bundle.ComponentPackage, set *bundle.ComponentSe
 }
 
 // Validate validates components and components sets, providing as many errors as it can.
-func (v *ComponentValidator) Validate() []error {
-	var errs []error
+func (v *ComponentValidator) Validate() field.ErrorList {
```

Comment:
What's the advantage of wrapping the things we want to validate in a `ComponentValidator`, vs. just having a validation function per type?

Reply:
Good question. I think it's just ignorance on my part. What's the standard way to add these validation functions in K8S? Is it in the API directory? That's not ideal, because it'll lead to circular deps at the moment.

Reply:
Yeah basically `Validate{Type}(obj *Type) field.ErrorList` functions for top-level resource types (usually that's anything that has ObjectMeta), and you can also have `validate{EmbeddedType}(obj *EmbeddedType, p *field.Path) field.ErrorList` style helpers for substructures that let you thread the fieldpath through validation. 

If you always start from validating a root object where every field has a name (even implicit), you also don't have to worry about subscripting an anonymous field (from above comment), because you can just include the name in the path without ambiguity (comment above on minimum I think we could do if you can't do this).

Reply:
Conventionally you would put the validation close to where the objects are defined. What are the circular deps at the moment?

Reply:
I kept around the validator object. I'm imagining that I'll want configurable validation, so it's nice to have this around. That being the case, I feel like having a single validator object is more flexible.